### PR TITLE
Create test environment for emme assignment

### PR DESCRIPTION
**Pull request by zptro**
Created mock-up version of `EmmeProject`. Emulates key features of EMME API, so that `EmmeAssignmentModel` and `AssignmentPeriod` can be unit tested without EMME binding.

### CODE REVIEW

File: Scripts/assignment/emme_bindings/mock_project.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+
+
+class MockProject:
+    """Mock-up version of `EmmeProject`.
+
+    Emulates key features of EMME API, so that `EmmeAssignmentModel`
+    and `AssignmentPeriod` can be unit tested without EMME binding.
+    """
+
+    path = "c:\\xxx\\yyy"
```
Comment (johpiip) [inline Scripts/assignment/emme_bindings/mock_project.py:12]: What is the meaning of this `path`?

**Review by johpiip — APPROVED** [review]
Looks good to me! VSCode finds new tests and they give no errors. Just checking: is all the code in `mock_project.py` written by us and not copied from the real classes?

I came across "old-style and new-style classes" when googling for differences between Python 2 and 3. [Here](https://portingguide.readthedocs.io/en/latest/classes.html) is an example. Do you think this will be a problem later?

Edit. Old-style: `class OldStyle: pass` and new-style: `class NewStyle(object)`.

**Comment by johpiip** [comment]
Additional review comment, I only notices this now! Results are printed to `tests/test_data/Results/2016_test` instead of `tests/test_data/Results/test` where all other results go. That folder is not gitignored so they clutter git status.

**Comment by zptro** [comment]
> Looks good to me! VSCode finds new tests and they give no errors. Just checking: is all the code in `mock_project.py` written by us and not copied from the real classes?

Yes, all written by us based on the Emme documentation. The real classes are (I think) compiled and cannot be inspected.
 
> I came across "old-style and new-style classes" when googling for differences between Python 2 and 3. [Here](https://portingguide.readthedocs.io/en/latest/classes.html) is an example. Do you think this will be a problem later?

These classes do not inherit anything from `object`. There are only a few such classes in Helmet.

> Additional review comment, I only notices this now! Results are printed to tests/test_data/Results/2016_test instead of tests/test_data/Results/test where all other results go. That folder is not gitignored so they clutter git status.

Good point! I will fix this!

File: Scripts/assignment/emme_bindings/mock_project.py (hunk 1 above)
Comment (zptro) [reply Scripts/assignment/emme_bindings/mock_project.py:12]: In `EmmeProject`, it refers to the directory path of the *emmebank* file and it is used to locate the function files here: https://github.com/HSLdevcom/helmet-model-system/blob/8d6073468da8c7519fd575edbd809e5804219707/Scripts/assignment/assignment_period.py#L460

I will remove this dependency in the Helmet 4.1 development.

**Comment by johpiip** [comment]
> These classes do not inherit anything from `object`. There are only a few such classes in Helmet.

But they will inherit in Python 3, implicitly, based on the docs. Well, this is just one thing to look at when moving from Python 2 to 3. 😄

**Review by johpiip — APPROVED** [review]
We discussed this on Monday. Looks good to me!

**Pull request merged**